homeassistant/components/dsmr_reader/config_flow.py

```diff
@@ -0,0 +1,50 @@
+"""Config flow to configure DSMR Reader."""
+from __future__ import annotations
+
+from collections.abc import Awaitable
+import logging
+from typing import Any
+
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.config_entry_flow import DiscoveryFlowHandler
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def _async_has_devices(_: HomeAssistant) -> bool:
+    """Return true as this integration doesn't support any real devices."""
+    return True
+
+
+class DsmrReaderFlowHandler(DiscoveryFlowHandler[Awaitable[bool]], domain=DOMAIN):
+    """Handle DSMR Reader config flow. The MQTT step is inherited from the parent class."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Set up the config flow."""
+        super().__init__(DOMAIN, "DSMR Reader", _async_has_devices)
+
+    async def async_step_import(self, _: dict[str, Any] | None) -> FlowResult:
+        """Import from configuration.yaml and create config entry."""
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+        if not self.hass.services.has_service(domain="mqtt", service="publish"):
+            _LOGGER.warning("DSMR Reader configuration import failed. MQTT is missing")
+            return self.async_abort(reason="mqtt_missing")
```

Comment:
MQTT is already a dependency of this integration. The config flow will not start unless MQTT is set up.
```suggestion
```

Reply:
If I remove this check, I get the error below when someone has the dsmr_reader added as sensor in their configuration.yaml while they don't have MQTT configured. Admittedly, that is a very weird situation to be in, but at the same time it's such an easy thing to catch... I don't mind removing it if other reviewers concur or if I'm overlooking something else.

```
2022-07-02 09:41:41 ERROR (MainThread) [homeassistant.components.sensor] Error while setting up dsmr_reader platform for sensor
Traceback (most recent call last):
  File "/workspaces/core/homeassistant/helpers/entity_platform.py", line 257, in _async_setup_platform
    await asyncio.gather(*pending)
  File "/workspaces/core/homeassistant/helpers/entity_platform.py", line 382, in async_add_entities
    await asyncio.gather(*tasks)
  File "/workspaces/core/homeassistant/helpers/entity_platform.py", line 619, in _async_add_entity
    await entity.add_to_platform_finish()
  File "/workspaces/core/homeassistant/helpers/entity.py", line 809, in add_to_platform_finish
    await self.async_added_to_hass()
  File "/workspaces/core/homeassistant/components/dsmr_reader/sensor.py", line 74, in async_added_to_hass
    await mqtt.async_subscribe(
  File "/workspaces/core/homeassistant/components/mqtt/__init__.py", line 544, in async_subscribe
    async_remove = await hass.data[DATA_MQTT].async_subscribe(
KeyError: 'mqtt'
```

Reply:
I've removed the mqtt check, as requested. I guess it is unlikely someone wouldn't have mqtt setup when a configuration is being imported anyway.